### DISCUSSION TITLE
How to Exclude Specific Keyword from Identifier in Tree-sitter Grammar?

### DISCUSSION THREAD

**Discussion by sho94os** [tree-sitter/tree-sitter#3607] [Q&A] [answered]
any one can help please

this is my langauge reserved word

`    again_keyword: $ => /again/i,
    and_keyword: $ => /and/i,
    but_keyword: $ => /but/i,
    bye_keyword: $ => /bye/i,
    call_keyword: $ => /call/i,
    case_keyword: $ => /case/i,
    catch_keyword: $ => /catch/i,
    class_keyword: $ => /class/i,
    def_keyword: $ => /def/i,
    do_keyword: $ => /do/i,
    done_keyword: $ => /done/i,
    else_keyword: $ => /else/i,
    elseif_keyword: $ => /elseif/i,
    end_keyword: $ => /end/i,
    exit_keyword: $ => /exit/i,
    for_keyword: $ => /for/i,
    foreach_keyword: $ => /foreach/i,
    func_keyword: $ => /func/i,
    from_keyword: $ => /from/i,
    get_keyword: $ => /get/i,
    give_keyword: $ => /give/i,
    if_keyword: $ => /if/i,
    import_keyword: $ => /import/i,
    in_keyword: $ => /in/i,
    load_keyword: $ => /load/i,
    loop_keyword: $ => /loop/i,
    new_keyword: $ => /new/i,
    next_keyword: $ => /next/i,
    not_keyword: $ => /not/i,
    off_keyword: $ => /off/i,
    ok_keyword: $ => /ok/i,
    on_keyword: $ => /on/i,
    or_keyword: $ => /or/i,
    other_keyword: $ => /other/i,
    package_keyword: $ => /package/i,
    private_keyword: $ => /private/i,
    put_keyword: $ => /put/i,
    return_keyword: $ => /return/i,
    see_keyword: $ => /see/i,
    step_keyword: $ => /step/i,
    switch_keyword: $ => /switch/i,
    to_keyword: $ => /to/i,
    try_keyword: $ => /try/i,
    while_keyword: $ => /while/i,
    endfunc_keyword: $ => /endfunc/i,
    endclass_keyword: $ => /endclass/i,
    endpackage_keyword: $ => /endpackage/i,
    endif_keyword: $ => /endif/i,
    endfor_keyword: $ => /endfor/i,
    endwhile_keyword: $ => /endwhile/i,
    endswitch_keyword: $ => /endswitch/i,
    endtry_keyword: $ => /endtry/i,
    function_keyword: $ => /function/i,
    endfunction_keyword: $ => /endfunction/i,
    break_keyword: $ => /break/i,
    continue_keyword: $ => /continue/i,

    identifier: $ => /[a-zA-Z]+/`

**Comment by amaanq** [comment]
It needs #1635, which is not a thing yet

**Reply by amaanq** [reply, marked as answer]
update, this *is* now a thing (or will be in 0.25.0) - #3896

**Reply by MaximusSeniorem** [reply]
is this a thing then ?

I reeeeaaaaaaaaaally need the feature (I don't have enough 'a's and 'e's in order to emphasize the need), in order to limit errors propagation. 

I saw it was merged in the 0.24.7 release, using web-tree-sitter 0.24.7 it does not seem to work.

**Reply by clason** [reply]
It was not; features are not backported to the release-0.24 branch. You need HEAD or the upcoming 0.25.0 release.

**Reply by MaximusSeniorem** [reply]
works like a charm. thanks a lot
